src/Microsoft.ApiDesignGuidelines.Analyzers/Core/PropertiesShouldNotBeWriteOnly.cs

```diff
@@ -36,14 +35,45 @@ public abstract class PropertiesShouldNotBeWriteOnlyAnalyzer : DiagnosticAnalyze
                                                                              DiagnosticSeverity.Warning,
                                                                              isEnabledByDefault: false,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/ms182165.aspx",
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(AddGetterRule, MakeMoreAccessibleRule);
 
         public override void Initialize(AnalysisContext analysisContext)
         {
-            
+            analysisContext.RegisterSymbolAction(symbolContext => AnalyzeSymbol(symbolContext), SymbolKind.Property);
+        }
+
+        /// <summary>
+        /// Implementation for CA1044: Properties should not be write only
+        /// </summary>
+        /// <param name="context"></param>
+        private static void AnalyzeSymbol(SymbolAnalysisContext context)
+        {
+
+            var property = context.Symbol as IPropertySymbol;
+            // If no property found
+            if (property == null)
+            {
+                return;
+            }
+            // check for when rule is turned off
```

Comment:
Yes.  That’s what I meant…Will look at your comments and fix later today.

Thanks, Annie
From: Srivatsn Narayanan [mailto:notifications@github.com]
Sent: Monday, February 1, 2016 10:04 AM
To: dotnet/roslyn-analyzers <roslyn-analyzers@noreply.github.com>
Cc: Annie Poh <anniepoh@microsoft.com>
Subject: Re: [roslyn-analyzers] CA1044 Implementation and Unit Test (#827)


In src/Microsoft.ApiDesignGuidelines.Analyzers/Core/PropertiesShouldNotBeWriteOnly.cs<https://github.com/dotnet/roslyn-analyzers/pull/827#discussion_r51454686>:

> +        }

> +

> +        /// <summary>

> +        /// Implementation for CA1044: Properties should not be write only

> +        /// </summary>

> +        /// <param name="context"></param>

> +        private static void AnalyzeSymbol(SymbolAnalysisContext context)

> +        {

> +

> +            var property = context.Symbol as IPropertySymbol;

> +            // If no property found

> +            if (property == null)

> +            {

> +                return;

> +            }

> +            // check for when rule is turned off

Can you clarify this comment - I think what you are trying to say is that you don't want to issue a diagnostic for overridden properties.

—
Reply to this email directly or view it on GitHub<https://github.com/dotnet/roslyn-analyzers/pull/827/files#r51454686>.